src/python-common/ceph/deployment/service_spec.py

```diff
@@ -395,10 +395,22 @@ def from_json(cls, json_spec):
         :param json_spec: A valid dict with ServiceSpec
         """
 
-        service_type = json_spec.get('service_type', '')
+        c = json_spec.copy()
+
+        # kludge to make `from_json` compatible to `Orchestrator.describe_service`
+        # Open question: Remove `service_id` form to_json?
```

Comment:
i think yes... if service_name is always present then service_type is redundant

Reply:
hm. rethinking this. I actually like `service_type`, as it clearly states the type of the spec.